code/modules/shuttle/cas_plane.dm

```diff
@@ -308,13 +308,11 @@
 		return
 	if(!GLOB.cameranet.checkTurfVis(get_turf_pixel(target)))
 		return
-	if(!active_weapon)
-		to_chat(source, "<span class='warning'>No active weapon selected!</span>")
-		return
 	var/area/A = get_area(target)
 	if(A.ceiling >= CEILING_DEEP_UNDERGROUND)
-		to_chat(source, "<span class='warning'>That target is too deep underground!</span>")
-		return
+		if(!active_weapon.ammo_equipped.Penetrate_Cave == TRUE)
```

Comment:
1) penetrate_cave is already a boolean, no need to do if (TRUE == TRUE)
2) you tested this? looks like the boolean math is wrong. And even if it's doing the correct thing, this is very much shit code, redo it please

Reply:
If by shitcode, you mean I shouldn't have that as a  == TRUE. Yes, Ill go fix that. If not, please elaborate.

Reply:
You did !a==b when you wanted to do !(a==b) or (a!=b). The first one you did was only working because penetrate_cave was a boolean, it will not work in general

Reply:
Ah, well I removed the ==TRUE so that should fix it